src/Detector/ParserHelper.cs

```diff
@@ -31,7 +32,17 @@ public static TomlTable ParseTomlFile(ISourceRepo sourceRepo, string filePath)
 
             try
             {
-                return Toml.ReadString(tomlContent);
+                // Gets a syntax tree of the TOML text
+                var doc = Toml.Parse(tomlContent); // returns a DocumentSyntax
+                // Check for parsing errors with doc.HasErrors and doc.Diagnostics
+                // doc.HasErrors => throws an exception
+                if (doc.HasErrors)
```

Comment:
Is there a way to get list of errors that this parser gives back? if yes, are those errors captured in an exception? if yes we want to throw that in a new FailedToParseFileException because we want to know what is the actual error that the parser faced and not the following generic message because it does not give us an idea of what the problem is with the file.

Reply:
I see, yeah it has a diagnostics property which contains detail error information, I didn't try it yet, I'll try it and if it works, then pass the details into `FailedToParseFileException `.